# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerRegistration.h
# Commit: 789ce1a19e9f
# Full Hash: 789ce1a19e9f09a806dde2a27480963f45832218
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2021-06-10 09:49:36
# Regressor Bug: 1564235
# File Overlap Count: 3
# Description:
#   Bug 1564235 - P2 Create NavigationPreloadManager.webidl and implement webidl binding. r=dom-worker-reviewers,webidl,asuth
#   
#   This patch creates NavigationPreloadManager.webidl and implements the webidl binding.
#   
#   Depends on D116913
# ==============================================================================

diff -r a4cd096b0ba7 -r 789ce1a19e9f dom/serviceworkers/ServiceWorkerRegistration.h
--- a/dom/serviceworkers/ServiceWorkerRegistration.h	Thu Jun 10 00:48:10 2021 +0000
+++ b/dom/serviceworkers/ServiceWorkerRegistration.h	Thu Jun 10 00:48:11 2021 +0000
@@ -21,6 +21,7 @@
 namespace mozilla {
 namespace dom {
 
+class NavigationPreloadManager;
 class Promise;
 class PushManager;
 class WorkerPrivate;
@@ -81,6 +82,8 @@
 
   already_AddRefed<ServiceWorker> GetActive() const;
 
+  already_AddRefed<NavigationPreloadManager> NavigationPreload();
+
   void UpdateState(const ServiceWorkerRegistrationDescriptor& aDescriptor);
 
   bool MatchesDescriptor(
@@ -133,6 +136,7 @@
   RefPtr<ServiceWorker> mInstallingWorker;
   RefPtr<ServiceWorker> mWaitingWorker;
   RefPtr<ServiceWorker> mActiveWorker;
+  RefPtr<NavigationPreloadManager> mNavigationPreloadManager;
   RefPtr<PushManager> mPushManager;
 
   struct VersionCallback {